apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -487,9 +489,11 @@ public override Task<PooledSession> WithFreshTransactionOrNewAsync(PooledSession
             /// <summary>
             /// Release a session back to the pool (or refresh or evict it) and decrement the number of active sessions.
             /// </summary>
-            public override void Release(PooledSession session, bool deleteSession)
+            public override void Release(PooledSession session, ByteString transactionToRollback, bool deleteSession)
             {
                 Interlocked.Decrement(ref _activeSessionCount);
+                Parent.MaybeRollbackTransactionFireAndForget(session.SessionName, transactionToRollback);
```

Comment:
Can't we add this call in EvictSession and ReleaseInactiveSession instead. Rolling back is something that we'll want to attempt eitherway so it should be part of it (Evict or Releasing) I think.
And we don't have to remember to add the calls to MaybeRollback... whenever we Evict or Release.

Reply:
We don't always. I originally put it in there, but there are cases where we call ReleaseInactiveSession but don't want to roll anything back... e.g. because we've just allocated the transaction in TryCreateReadWriteTransactionAndReturnToPool.

Reply:
Yep, fine as well then.